### PR TITLE
[#IOCOM-921] new definitions for the remote content configuration apis

### PR DESCRIPTION
**Pull request by acuiuli**
#### List of Changes

- [new definitions for the remote content configuration apis](https://github.com/pagopa/io-functions-commons/pull/358/commits/ff121cb9635b308ab09c44d80940e887fa549c9b)
- [fixed string field with minLength for rc configuration](https://github.com/pagopa/io-functions-commons/pull/358/commits/ad90df72b7840f325e5f19ad62831e355f1cba77)
- [small fix](https://github.com/pagopa/io-functions-commons/pull/358/commits/a044867dd99d53e5eb29e44eed8950a96b874397)

#### Motivation and Context

Added definitions to manage the model of the remote content configuration and the response of the GetRCConfiguration api

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: openapi/definitions.yaml [hunk 1]
```diff
@@ -544,6 +546,139 @@ InternalMessageResponseWithContent:
...
+    description:
+      type: string
+      description: Configuration description.
+    hasPrecondition:
+       $ref: "#/HasPrecondition"
+    disableLollipopFor:
+      type: array
+      items:
+        type: string
+        description: User for which lollipop shoulg be disabled
```
Comment (Garma00) [inline openapi/definitions.yaml:578]:
```suggestion
        description: User for which lollipop should be disabled
```

File: openapi/definitions.yaml [hunk 2]
```diff
@@ -544,6 +546,139 @@ InternalMessageResponseWithContent:
...
+  properties:
+    headerKeyName:
+      type: string
+      description: header key name
+    key:
+      type: string
+      description: authentication config key
+    type:
+      type: string
+      description: authentication config type
```
Comment (Garma00) [inline openapi/definitions.yaml:647]: Take care, those properties should be `NonEmptyString`, we need to add `minLength: 1`

**Review by Garma00 — CHANGES_REQUESTED** [review]
LGTM apart of the comment about the `NonEmptyString` type, check out all the properties marked as `string`

**Review by Garma00 — APPROVED** [review]
LGTM

**Review by michaeldisaro — APPROVED** [review]

**Pull request merged**